audit-implementation.md

```diff
@@ -17,7 +17,7 @@ Implementation:
 namespace App\Models;
 
 use Jenssegers\Mongodb\Eloquent\Model;
-use Jenssegers\Mongodb\Relations\MorphTo;
+use Illuminate\Database\Eloquent\Relations\MorphTo;
```

Comment:
To be honest, I never used MongoDB with PHP, but since the `Jenssegers\Mongodb\Relations\MorphTo` class extends  `Illuminate\Database\Eloquent\Relations\MorphTo`, I reckon it should work.

Do you get an error from PHP?

Reply:
No. It won't work.

```
<?php

class Child {}
class ChildB extends Child {}

abstract class A
{
    abstract public function foo(): Child;
}

class B extends A
{
    public function foo(): ChildB
    {
        return new ChildB;
    }
}
```
result: Declaration of B::foo(): ChildB must be compatible with A::foo(): Child on line 17

and correct example:

```
<?php

class Child {}
class ChildB extends Child {}

abstract class A
{
    abstract public function foo(): Child;
}

class B extends A
{
    public function foo(): Child
    {
        return new ChildB;
    }
}
```

So in your example of integration according to php logic, you do incorrect contract implementation.

Reply:
We are using **Jenssegers** implementation, it's not about implementation it's about an error in php related to return type. In your example, you trying to force return value to other than defined in the contract. Please look carefully into my example.

Reply:
I get that PHP blows up because of the return type. What I'm asking is, if you're using the `Jenssegers` implementation, shouldn't you also be using `Jenssegers\Mongodb\Relations\MorphTo` instead of `Illuminate\Database\Eloquent\Relations\MorphTo`, or it works just the same?

If the former, I'm thinking of making the contract a bit more flexible to allow any implementation.

Reply:
I using **Jenssegers\Mongodb\Relations\MorphTo**, it works just the same. So you don't need to do changes for anything in the code, only correct documentation. 

Btw, thank you for the great tool! ;-)

Reply:
Basically, what I want to know is if the extra/overriden methods [here](https://github.com/jenssegers/laravel-mongodb/blob/master/src/Jenssegers/Mongodb/Relations/MorphTo.php#L7) are of actual use, or not.

Reply:
Yes, they are in use, but I don't see any problems with this.

Reply:
For testing perpose it's easy to create a couple tests that testing mongo support in your lib.